### PR TITLE
Hw 1.2.0

### PR DESCRIPTION
**Pull request by hwvlapin**

### CODE REVIEW

File: src/EventListener/LogRevisionsListener.php [hunk 1]
```diff
@@ -200,7 +215,7 @@ public function postPersist(LifecycleEventArgs $eventArgs): void
         // onFlush was executed before, everything already initialized
         $entity = $eventArgs->getEntity();
 
-        $class = $this->em->getClassMetadata(\get_class($entity));
+        $class = $this->em->getClassMetadata(get_class($entity));
```
Comment (maMykola) [inline src/EventListener/LogRevisionsListener.php]: add `use function` statement for function `get_class` or revert back backslash

File: src/EventListener/LogRevisionsListener.php [hunk 2]
```diff
@@ -227,7 +242,7 @@ public function postUpdate(LifecycleEventArgs $eventArgs): void
         }
 
         // if we have no changes left => don't create revision log
-        if (0 === \count($changeset)) {
+        if (count($changeset) === 0) {
```
Comment (maMykola) [inline src/EventListener/LogRevisionsListener.php]: revert back condition statement

File: src/EventListener/LogRevisionsListener.php [hunk 3]
```diff
@@ -250,13 +265,13 @@ public function onFlush(OnFlushEventArgs $eventArgs): void
             //doctrine is fine deleting elements multiple times. We are not.
             $hash = $this->getHash($entity);
 
-            if (\in_array($hash, $processedEntities, true)) {
+            if (in_array($hash, $processedEntities)) {
```
Comment (maMykola) [inline src/EventListener/LogRevisionsListener.php]: add `use statement` for function `in_array` or revert back backslash

Comment (maMykola) [inline src/EventListener/LogRevisionsListener.php]: is there really need to remove third argument? as default value equals `false`

File: src/EventListener/LogRevisionsListener.php [hunk 4]
```diff
@@ -339,8 +354,8 @@ private function getInsertRevisionSQL($class)
             $placeholders = ['?', '?'];
             $tableName = $this->config->getTableName($class);
 
-            $sql = 'INSERT INTO '.$tableName.' ('.
-                $this->config->getRevisionFieldName().', '.$this->config->getRevisionTypeFieldName();
+            $sql = "INSERT INTO " . $tableName . " (" .
```
Comment (maMykola) [inline src/EventListener/LogRevisionsListener.php]: use single quotes instead of double

File: src/EventListener/LogRevisionsListener.php [hunk 5]
```diff
@@ -352,14 +367,14 @@ private function getInsertRevisionSQL($class)
...
-                        $sql .= ', '.$sourceCol;
+                        $sql .= ', ' . $sourceCol;
                         $placeholders[] = '?';
                     }
                 }
             }
 
             foreach ($class->fieldNames as $field) {
-                if (\array_key_exists($field, $fields)) {
+                if (array_key_exists($field, $fields)) {
```
Comment (maMykola) [inline src/EventListener/LogRevisionsListener.php]: add use statement for function or revert back backslash

File: src/EventListener/LogRevisionsListener.php [hunk 6]
```diff
@@ -374,17 +389,17 @@ private function getInsertRevisionSQL($class)
...
+            if (($class->isInheritanceTypeJoined() && $class->rootEntityName == $class->name)
                 || $class->isInheritanceTypeSingleTable()
             ) {
-                $sql .= ', '.$class->discriminatorColumn['name'];
+                $sql .= ', ' . $class->discriminatorColumn['name'];
                 $placeholders[] = '?';
             }
 
-            $sql .= ') VALUES ('.implode(', ', $placeholders).')';
+            $sql .= ") VALUES (" . implode(", ", $placeholders) . ")";
```
Comment (maMykola) [inline src/EventListener/LogRevisionsListener.php]: single quotes

File: src/EventListener/LogRevisionsListener.php [hunk 7]
```diff
@@ -414,15 +429,15 @@ private function saveRevisionEntityData($class, $entityData, $revType): void
             $data = $entityData[$field] ?? null;
             $relatedId = false;
 
-            if (null !== $data && $this->uow->isInIdentityMap($data)) {
+            if ($data !== null && $this->uow->isInIdentityMap($data)) {
```
Comment (maMykola) [inline src/EventListener/LogRevisionsListener.php]: revert back condition

File: src/EventListener/LogRevisionsListener.php [hunk 8]
```diff
@@ -414,15 +429,15 @@ private function saveRevisionEntityData($class, $entityData, $revType): void
...
                 $relatedId = $this->uow->getEntityIdentifier($data);
             }
 
             $targetClass = $this->em->getClassMetadata($assoc['targetEntity']);
 
             foreach ($assoc['sourceToTargetKeyColumns'] as $sourceColumn => $targetColumn) {
                 $fields[$sourceColumn] = true;
-                if (null === $data) {
+                if ($data === null) {
```
Comment (maMykola) [inline src/EventListener/LogRevisionsListener.php]: revert back condition

File: src/EventListener/LogRevisionsListener.php [hunk 9]
```diff
@@ -451,10 +466,11 @@ private function saveRevisionEntityData($class, $entityData, $revType): void
...
             $params[] = $class->discriminatorValue;
             $types[] = $class->discriminatorColumn['type'];
-        } elseif ($class->isInheritanceTypeJoined()
-            && $class->name === $class->rootEntityName
+        } elseif (
+            $class->isInheritanceTypeJoined() &&
+            $class->name === $class->rootEntityName
         ) {
-            $params[] = $entityData[$class->discriminatorColumn['name']];
+            $params[] = isset($entityData[$class->discriminatorColumn['name']]) ? $entityData[$class->discriminatorColumn['name']] : $class->discriminatorValue;
```
Comment (maMykola) [inline src/EventListener/LogRevisionsListener.php]: use ternary operation for default value if not set

File: src/EventListener/LogRevisionsListener.php [hunk 10]
```diff
@@ -512,7 +528,7 @@ private function prepareUpdateData($persister, $entity)
         $versionField = null;
         $result = [];
 
-        if (false !== ($versioned = $classMetadata->isVersioned)) {
+        if (($versioned = $classMetadata->isVersioned) !== false) {
```
Comment (maMykola) [inline src/EventListener/LogRevisionsListener.php]: revert back condition

File: src/EventListener/LogRevisionsListener.php [hunk 11]
```diff
@@ -541,15 +557,15 @@ private function prepareUpdateData($persister, $entity)
                 continue;
             }
 
-            if (null !== $newVal) {
+            if ($newVal !== null) {
```
Comment (maMykola) [inline src/EventListener/LogRevisionsListener.php]: revert back condition

File: src/EventListener/LogRevisionsListener.php [hunk 12]
```diff
@@ -541,15 +557,15 @@ private function prepareUpdateData($persister, $entity)
...
                 if ($uow->isScheduledForInsert($newVal)) {
                     $newVal = null;
                 }
             }
 
             $newValId = null;
 
-            if (null !== $newVal) {
+            if ($newVal !== null) {
```
Comment (maMykola) [inline src/EventListener/LogRevisionsListener.php]: revert back condition

**Review by maMykola — CHANGES_REQUESTED** [review]

File: src/EventListener/LogRevisionsListener.php (hunk 1 above)
Comment (hwvlapin) [reply src/EventListener/LogRevisionsListener.php]: Fixed

File: src/EventListener/LogRevisionsListener.php (hunk 2 above)
Comment (hwvlapin) [reply src/EventListener/LogRevisionsListener.php]: Fixed

File: src/EventListener/LogRevisionsListener.php (hunk 3 above)
Comment (hwvlapin) [reply src/EventListener/LogRevisionsListener.php]: Fixed

Comment (hwvlapin) [reply src/EventListener/LogRevisionsListener.php]: Fixed

File: src/EventListener/LogRevisionsListener.php (hunk 4 above)
Comment (hwvlapin) [reply src/EventListener/LogRevisionsListener.php]: Fixed

File: src/EventListener/LogRevisionsListener.php (hunk 8 above)
Comment (hwvlapin) [reply src/EventListener/LogRevisionsListener.php]: Fixed

File: src/EventListener/LogRevisionsListener.php (hunk 9 above)
Comment (hwvlapin) [reply src/EventListener/LogRevisionsListener.php]: Fixed

File: src/EventListener/LogRevisionsListener.php (hunk 10 above)
Comment (hwvlapin) [reply src/EventListener/LogRevisionsListener.php]: Fixed

File: src/EventListener/LogRevisionsListener.php (hunk 11 above)
Comment (hwvlapin) [reply src/EventListener/LogRevisionsListener.php]: Fixed

File: src/EventListener/LogRevisionsListener.php (hunk 12 above)
Comment (hwvlapin) [reply src/EventListener/LogRevisionsListener.php]: Fixed

**Review by maMykola — COMMENTED** [review]
There is a need to also move changes in `LogRevisionsListener`

**Review by maMykola — APPROVED** [review]

**Pull request merged**